<?xml version="1.0"?>
<launch>
    <!-- Followng Launche File instantiates Robot Comau Smart Six using 'comau_smart_six' as prefix of every
    sub-module, like TF_names or JointNames. Change robot name in XACRO file to remove this behaviour -->
    <arg name="robot_name" default="grasshopper"/>
    <arg name="viz" default="false"/>
    <arg name="ik_service" default="true"/>

    <group ns="$(arg robot_name)">

        <param name="robot_description" command="$(find xacro)/xacro.py '$(find rocup)/urdf/grasshopper/grasshopper.urdf.xacro'"/>

        <param name="use_gui" value="true"/>

        <node name="grasshopper_state_publisher" pkg="robot_state_publisher" type="state_publisher">
            <param name="publish_frequency" value="1000"/>
            <param name="aatf_prefix" value="$(arg robot_name)"/>
            <param name="joint_states" value="/grasshopper/joint_states"/>
        </node>

        <group if="$(arg ik_service)">
            <node pkg="rocup" type="grasshopper_ik_service" output="screen" name="ik_service_node">
                <param name="robot_name" value="$(arg robot_name)"/>
            </node>
        </group>

    </group>
</launch>